dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-email/src/main/java/org/apache/dolphinscheduler/plugin/alert/email/MailSender.java

```diff
@@ -378,13 +382,42 @@ private AlertResult getStringObjectMap(String title, String content, AlertResult
 
         // send
         email.setDebug(true);
-        email.send();
-
+        email.buildMimeMessage();
+        sendMail(email.getMimeMessage(), email.getMailSession());
         alertResult.setSuccess(true);
 
         return alertResult;
     }
 
+    /**
+     * send mail with validAddresses retry
+     *
+     * @param mimeMessage the message
+     * @param session connectSession
+     */
+    private void sendMail(MimeMessage mimeMessage, Session session) throws MessagingException, AlertEmailException {
+        Transport transport = session.getTransport(new SMTPProvider());
+        transport.addTransportListener(new TransportAdapter() {
+            @Override
+            public void messageNotDelivered(TransportEvent event) {
+                Address[] validUnsentAddresses = event.getValidUnsentAddresses();
+                if (validUnsentAddresses == null || validUnsentAddresses.length < 1) {
+                    return;
+                }
+                try {
+                    transport.sendMessage(mimeMessage, validUnsentAddresses);
+                } catch (MessagingException e) {
+                    log.error("send mail with validAddresses failed:{}", e.getMessage());
+                    throw new AlertEmailException("send mail with validAddresses failed", e);
```

Comment:
We don't need to throw exception here.

Reply:
ok

Reply:
@SbloodyS
mail sender remove inappropriate exception throwing